sdlib/d/gc/hpd.d

```diff
@@ -111,6 +111,46 @@ public:
 		return bestIndex;
 	}
 
+	bool reserve(uint index, uint pages) {
+		assert(pages > 0 && pages <= PageCount, "Invalid number of pages!");
+		assert(index <= PageCount - pages, "Invalid index!");
+
+		if (pages > longestFreeRange) {
+			return false;
+		}
+
+		uint freePos, freeLength;
+		allocatedPages.nextFreeRange(index, freePos, freeLength);
```

Comment:
This doesn't really do what we want. We are not interested in the next free range. We are interested to know if there is a free range of a certain size right there.